src/gui/shellwidget/test/CMakeLists.txt

```diff
@@ -6,12 +6,16 @@ endif ()
 
 function(add_xtest SOURCE_NAME)
 	add_executable(${SOURCE_NAME} ${SOURCE_NAME}.cpp)
+	file(TO_NATIVE_PATH ${CMAKE_CURRENT_SOURCE_DIR} TEST_SOURCE_DIR)
```

Comment:
Move this line out of this function, before the add_definition.

The segfault happens because the test fails to open the file. Since TEST_SOURCE_DIR is no defined.